src/parsing.rs

```diff
@@ -176,11 +176,19 @@ fn parse_type_choices(
     }
 }
 
-fn type2_to_number_literal(type2: &Type2) -> isize {
+fn type2_to_number_literal(type2: &Type2) -> i128 {
     match type2 {
-        Type2::UintValue { value, .. } => *value as isize,
-        Type2::IntValue { value, .. } => *value,
-        Type2::FloatValue { value, .. } => *value as isize,
+        Type2::UintValue { value, .. } => *value as i128,
+        Type2::IntValue { value, .. } => *value as i128,
+        Type2::FloatValue { value, .. } => {
+            // FloatToInt trait still experimental so just directly check
```

Comment:
I'm guessing this was more of something copy-pasted that went unnoticed during the float PR that someone made. I don't think we'd want silent converting to integers as the preferred behavior there.